misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java

```diff
@@ -354,13 +427,16 @@ private void executeBeforeEachFunction(Test test, TestSuite suite, ClassLoader c
             suite.getBeforeEachFunctionNames().forEach(beforeEachTest -> {
                 String errorMsg;
                 try {
-                    invokeTestFunction(suite, beforeEachTest, classLoader, scheduler);
+                    Object value = invokeTestFunction(suite, beforeEachTest, classLoader, scheduler);
+                    if (value instanceof BError || value instanceof  Exception || value instanceof Error) {
+                        throw (Throwable) value;
+                    }
                 } catch (Throwable e) {
                     shouldSkip.set(true);
                     errorMsg = String.format("\t[fail] " + beforeEachTest +
-                                                     " [before each test function for the test %s] :\n\t    %s",
-                                             test,
-                                             formatErrorMessage(e));
+                                    " [before each test function for the test %s] :\n\t    %s",
+                            test,
+                            formatErrorMessage(e));
```

Comment:
I notice that the 'formatErrorMessage' method can throw an exception at some cases. If an exception gets thrown, it will not print the actual error message to the error stream(done in the following line).
This is problematic. Shall we handle the exception within the 'formatErrorMessage' method and return a formatted error message? WDYT?

Reply:
Instead of throwing the exception, we are returning the string version to be printed in the console